modules/scicos_blocks/src/c/cmatview.c

```diff
@@ -175,6 +184,35 @@ SCICOS_BLOCKS_IMPEXP void cmatview(scicos_block * block, scicos_flag flag)
             u = GetRealInPortPtrs(block, 1);
 
             result = pushData(block, u);
+            int i,j;
+            int m, n;
+            double alpha, beta ,min ,max;
+            double *scaledData;
+
+            iAxeUID = getAxe(iFigureUID, block);
+            iGrayplotUID = getGrayplot(iAxeUID, block);
+
+            m = GetInPortSize(block, 1, 1);
+            n = GetInPortSize(block, 1, 2);
+            /*
+            * Scale the data
+            */
+            alpha = block->rpar[0];
+            beta = block->rpar[1];
+            scaledData = (double *)MALLOC(m * n * sizeof(double));
+            if (scaledData == NULL)
+            {
+                result = FALSE;
+                return result;
+            }
+            fprintf(filePointer, "%d %d || %d | %d | %d | %d  | %d ||", block_id,processId, iFigureUID, iAxeUID,iGrayplotUID,m,n);
+            for (i = 0; i < m * n; i++)
+            {
+                scaledData[i] = floor(alpha * u[i] + beta);
+                fprintf(filePointer, " %f", scaledData[i]); // this values when passed to grayplot gives same chart as cmatview chart
```

Comment:
use scaledData instead of scaledData[i]

check if round() is better than floor()

for formatting, use %.0f instead of %f